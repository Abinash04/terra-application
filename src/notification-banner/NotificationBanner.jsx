import React from 'react';
import PropTypes from 'prop-types';
import uuidv4 from 'uuid/v4';

import BannerRegistrationContext from './private/BannerRegistrationContext';
import { BANNER_TYPES } from './private/utils';

const propTypes = {
  /**
   * The text and corresponding callback to populate the action button of the banner.
   */
  bannerAction: PropTypes.shape({
    /**
     * The text to display in the banner button.
     */
    text: PropTypes.string,
    /**
     * The Callback function triggered when the action button is clicked. No parameters are passed.
     */
    onClick: PropTypes.func,
  }),
  /**
   * The message content to display in the banner.
   */
  description: PropTypes.node,
  /**
   * Callback function triggered when the dismiss button is clicked. The presence of this prop will cause
   * the dismiss button to be included on the banner. No parameters are passed.
   */
  onRequestClose: PropTypes.func,
  /**
   * The type of alert to be rendered. One of `alert`, `error`, `warning`, `unsatisfied`, `unverified` or `advisory`.
   */
  type: PropTypes.oneOf([
    BANNER_TYPES.ALERT,
    BANNER_TYPES.ERROR,
    BANNER_TYPES.WARNING,
    BANNER_TYPES.UNSATISFIED,
    BANNER_TYPES.UNVERIFIED,
    BANNER_TYPES.ADVISORY,
  ]).isRequired,
};

const NotificationBanner = ({
  bannerAction, description, onRequestClose, type,
}) => {
  /**
   * A unique identifier is generated for each Banner during construction. This will be used to
   * uniquely register/unregister the banner with ancestor Banner Managers without requiring consumers to
   * define unique identifiers themselves.
   */
  const uuid = React.useRef(uuidv4());
  const bannerRegistration = React.useContext(BannerRegistrationContext);

  React.useEffect(() => {
    /**
     * If the bannerRegistration value is the ProviderRegistrationContext's default value,
     * then there is not a matching BannerProvider above it in the hierarchy.
     * This is possible but likely not intentional, so the component warns.
     */
    if (!bannerRegistration && process.env.NODE_ENV !== 'production') {
      // eslint-disable-next-line no-console
      console.warn('A NotificationBanner was not rendered within the context of a NotificationBannerProvider. If this is unexpected, validate that the expected version of the terra-application package is installed.');
    }

    if (bannerRegistration && bannerRegistration.registerNotificationBanner) {
      bannerRegistration.registerNotificationBanner(uuid.current, {
        bannerAction,
        description,
        key: uuid.current,
        onRequestClose,
        type,
      });
    }
  }, [bannerRegistration, description, bannerAction, onRequestClose, type]);

  React.useEffect(() => () => {
    if (bannerRegistration && bannerRegistration.unregisterNotificationBanner) {
      bannerRegistration.unregisterNotificationBanner(uuid.current, type);
    }
  }, [bannerRegistration, type]);

  return null;
};

NotificationBanner.propTypes = propTypes;

export default NotificationBanner;
