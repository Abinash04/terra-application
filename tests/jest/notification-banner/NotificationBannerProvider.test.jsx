import React, { useState } from 'react';
import { act } from 'react-dom/test-utils';
import { render, fireEvent } from '@testing-library/react';
import 'terra-base';
import { IntlProvider } from 'react-intl';

import BannerRegistrationContext from '../../../src/notification-banner/private/BannerRegistrationContext';
import NotificationBanner, { NotificationBannerProvider } from '../../../src/notification-banner';
// eslint-disable-next-line import/no-unresolved, import/extensions
import { messages } from '../../../aggregated-translations/en'; // aggregation is pre-jest step so this will exist

jest.mock('terra-base');
global.console.warn = jest.fn();

let keyValue = 0;
const mockBannerProps = { type: 'error' };

// eslint-disable-next-line react/prop-types
const ChildContent = ({ showBannerOnRender = false, buttonId = 'show-banner', children = null }) => {
  const [hasBanner, setHasBanner] = useState(showBannerOnRender);
  keyValue += 1;
  return (
    <div>
      <button
        type="button"
        data-testid={buttonId}
        onClick={() => setHasBanner(!hasBanner)}
      >
        Toggle Notification Banner
      </button>
      {hasBanner ? <NotificationBanner {...mockBannerProps} key={keyValue} /> : null}
      {children}
    </div>
);
};

const renderComponentWithChild = (childrenContent, ProviderProps = {}) => {
  let childContext;

  const component = render(
    <IntlProvider locale="en" messages={messages}>
      <NotificationBannerProvider {...ProviderProps}>
        <BannerRegistrationContext.Consumer>
          {(context) => {
            childContext = context;
            return childrenContent;
          }}
        </BannerRegistrationContext.Consumer>
      </NotificationBannerProvider>
    </IntlProvider>,
  );

  return { component, context: childContext };
};

describe('NotificationBannerProvider', () => {
  beforeEach(() => {
    jest.resetAllMocks;
    // eslint-disable-next-line no-console
    console.warn.mockClear();
  });

  it('renders with children', () => {
    const { component } = renderComponentWithChild(null);
    expect(component.container).toMatchSnapshot();
  });

  it('renders with children', () => {
    const { component } = renderComponentWithChild(<div>child</div>);

    expect(component.container).toMatchSnapshot();
  });

  it('renders with children that has a an banner', () => {
    const { component } = renderComponentWithChild(<ChildContent showBannerOnRender />);

    expect(component.container).toMatchSnapshot();
  });

  describe('registerNotificationBanner', () => {
    it('provides registerNotificationBanner callback in context', () => {
      const { context } = renderComponentWithChild();

      expect(context).toBeDefined();
      expect(context).toHaveProperty('registerNotificationBanner', expect.any(Function));
    });

    it('does not register banner if no id is provided', () => {
      const { component, context } = renderComponentWithChild();
      expect(component.container.querySelectorAll('[data-terra-application-notification-banner]')).toHaveLength(0);

      context.registerNotificationBanner(undefined, mockBannerProps);

      // eslint-disable-next-line no-console
      expect(console.warn).toHaveBeenCalledWith('A banner cannot be registered without an identifier.');
      expect(component.container.querySelectorAll('[data-terra-application-notification-banner]')).toHaveLength(0);
    });

    it('registers banner when id is provided', () => {
      const { component, context } = renderComponentWithChild();
      expect(component.container.querySelectorAll('[data-terra-application-notification-banner]')).toHaveLength(0);

      act(() => {
        context.registerNotificationBanner('mockID', mockBannerProps);
      });

      // eslint-disable-next-line no-console
      expect(console.warn).not.toHaveBeenCalledWith('A banner cannot be registered without an identifier.');
      expect(component.container.querySelectorAll('[data-terra-application-notification-banner]')).toHaveLength(1);
    });

    it('registers banner when rendered after mount', () => {
      const { component } = renderComponentWithChild(<ChildContent />);

      expect(component.container).toMatchSnapshot();

      const childButton = component.getByTestId('show-banner');
      act(() => {
        fireEvent.click(childButton);
      });

      expect(component.container.querySelectorAll('[data-terra-application-notification-banner]')).toHaveLength(1);
      expect(component.container).toMatchSnapshot();
    });
  });

  describe('unregisterNotificationBanner', () => {
    it('provides unregisterNotificationBanner callback in context', () => {
      const { context } = renderComponentWithChild();

      expect(context).toBeDefined();
      expect(context).toHaveProperty('unregisterNotificationBanner', expect.any(Function));
    });

    it('does not unregister notification banner if no id is provided', () => {
      const { component, context } = renderComponentWithChild(<ChildContent showBannerOnRender />);

      // eslint-disable-next-line no-console
      console.warn.mockClear(); // clear terra-responsive-element will be deprecated warning

      expect(component.container.querySelectorAll('[data-terra-application-notification-banner]')).toHaveLength(1);

      context.unregisterNotificationBanner(undefined);

      // eslint-disable-next-line no-console
      expect(console.warn).toHaveBeenCalledWith('A banner cannot be unregistered without an identifier or banner type.');
      expect(component.container.querySelectorAll('[data-terra-application-notification-banner]')).toHaveLength(1);
    });

    it('does not unregister notification banner if banner type is not provided', () => {
      const { component, context } = renderComponentWithChild(<ChildContent showBannerOnRender />);

      // eslint-disable-next-line no-console
      console.warn.mockClear(); // clear terra-responsive-element will be deprecated warning

      expect(component.container.querySelectorAll('[data-terra-application-notification-banner]')).toHaveLength(1);

      context.unregisterNotificationBanner('mockID', undefined);

      // eslint-disable-next-line no-console
      expect(console.warn).toHaveBeenCalledWith('A banner cannot be unregistered without an identifier or banner type.');
      expect(component.container.querySelectorAll('[data-terra-application-notification-banner]')).toHaveLength(1);
    });

    it('unregisters notification banner when id is provided', () => {
      const { component } = renderComponentWithChild(<ChildContent showBannerOnRender />);
      expect(component.container.querySelectorAll('[data-terra-application-notification-banner]')).toHaveLength(1);

      expect(component.container).toMatchSnapshot();

      const childButton = component.getByTestId('show-banner');
      act(() => {
        fireEvent.click(childButton);
      });

      expect(component.container.querySelectorAll('[data-terra-application-notification-banner]')).toHaveLength(0);
      expect(component.container).toMatchSnapshot();
    });
  });

  describe(('Nested Notification Banner Provider'), () => {
    it('registers notfication banner with top-level Banner Provider', () => {
      const childComponent = (
        <ChildContent>
          <NotificationBannerProvider>
            <ChildContent buttonId="show-2nd-banner" />
          </NotificationBannerProvider>
        </ChildContent>
      );
      const { component } = renderComponentWithChild(childComponent);
      expect(component.container).toMatchSnapshot();

      // trigger top Provider's banner
      const childButton = component.getByTestId('show-banner');
      act(() => {
        fireEvent.click(childButton);
      });

      expect(component.container.querySelectorAll('[data-terra-application-notification-banner]')).toHaveLength(1);
      expect(component.container).toMatchSnapshot();

      // trigger nested banner Provider's banner
      const secondChildButton = component.getByTestId('show-2nd-banner');
      act(() => {
        fireEvent.click(secondChildButton);
      });

      expect(component.container.querySelectorAll('[data-terra-application-notification-banner]')).toHaveLength(2);
      expect(component.container).toMatchSnapshot();
    });

    it('unregisters banner with top-level Banner Provider', () => {
      const childComponent = (
        <ChildContent showBannerOnRender>
          <NotificationBannerProvider>
            <ChildContent buttonId="show-2nd-banner" showBannerOnRender />
          </NotificationBannerProvider>
        </ChildContent>
      );
      const { component } = renderComponentWithChild(childComponent);

      expect(component.container.querySelectorAll('[data-terra-application-notification-banner]')).toHaveLength(2);

      expect(component.container).toMatchSnapshot();

      // remove top Provider's banner
      const childButton = component.getByTestId('show-banner');
      act(() => {
        fireEvent.click(childButton);
      });

      expect(component.container.querySelectorAll('[data-terra-application-notification-banner]')).toHaveLength(1);
      expect(component.container).toMatchSnapshot();

      // remove nested banner Provider's banner
      const secondChildButton = component.getByTestId('show-2nd-banner');
      act(() => {
        fireEvent.click(secondChildButton);
      });

      expect(component.container.querySelectorAll('[data-terra-application-notification-banner]')).toHaveLength(0);
      expect(component.container).toMatchSnapshot();
    });
  });
});
