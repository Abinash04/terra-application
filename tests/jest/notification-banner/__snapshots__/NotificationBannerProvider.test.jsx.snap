// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationBannerProvider Nested Notification Banner Provider registers notfication banner with top-level Banner Provider 1`] = `
<div>
  <div
    class="content-container-fill"
  >
    <div
      class="header"
    />
    <div
      class="main"
    >
      <div
        class="scroll normalizer"
      >
        <div>
          <button
            data-testid="show-banner"
            type="button"
          >
            Toggle Notification Banner
          </button>
          <div
            class="content-container-fill"
          >
            <div
              class="header"
            />
            <div
              class="main"
            >
              <div
                class="scroll normalizer"
              >
                <div>
                  <button
                    data-testid="show-2nd-banner"
                    type="button"
                  >
                    Toggle Notification Banner
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`NotificationBannerProvider Nested Notification Banner Provider registers notfication banner with top-level Banner Provider 2`] = `
<div>
  <div
    class="content-container-fill"
  >
    <div
      class="header"
    >
      <div
        class="alert-base error narrow"
        data-terra-application-banner="error"
        role="alert"
      >
        <div
          class="body body-std"
        >
          <span
            class="icon"
          >
            <svg
              aria-hidden="true"
              class="tui-Icon icon IconError"
              focusable="false"
              height="1em"
              viewBox="0 0 48 48"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M24 0c13.3 0 24 10.7 24 24S37.3 48 24 48 0 37.3 0 24C0 10.8 10.7 0 23.9 0h.1z"
                fill="#E50000"
              />
              <path
                d="M7 21.5h34v5H7v-5z"
                fill="#FFF"
              />
            </svg>
          </span>
          <div
            class="section"
          >
            <strong
              class="title"
            >
              <span>
                Error.
              </span>
            </strong>
          </div>
        </div>
      </div>
    </div>
    <div
      class="main"
    >
      <div
        class="scroll normalizer"
      >
        <div>
          <button
            data-testid="show-banner"
            type="button"
          >
            Toggle Notification Banner
          </button>
          <div
            class="content-container-fill"
          >
            <div
              class="header"
            />
            <div
              class="main"
            >
              <div
                class="scroll normalizer"
              >
                <div>
                  <button
                    data-testid="show-2nd-banner"
                    type="button"
                  >
                    Toggle Notification Banner
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`NotificationBannerProvider Nested Notification Banner Provider registers notfication banner with top-level Banner Provider 3`] = `
<div>
  <div
    class="content-container-fill"
  >
    <div
      class="header"
    >
      <div
        class="alert-base error narrow"
        data-terra-application-banner="error"
        role="alert"
      >
        <div
          class="body body-std"
        >
          <span
            class="icon"
          >
            <svg
              aria-hidden="true"
              class="tui-Icon icon IconError"
              focusable="false"
              height="1em"
              viewBox="0 0 48 48"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M24 0c13.3 0 24 10.7 24 24S37.3 48 24 48 0 37.3 0 24C0 10.8 10.7 0 23.9 0h.1z"
                fill="#E50000"
              />
              <path
                d="M7 21.5h34v5H7v-5z"
                fill="#FFF"
              />
            </svg>
          </span>
          <div
            class="section"
          >
            <strong
              class="title"
            >
              <span>
                Error.
              </span>
            </strong>
          </div>
        </div>
      </div>
    </div>
    <div
      class="main"
    >
      <div
        class="scroll normalizer"
      >
        <div>
          <button
            data-testid="show-banner"
            type="button"
          >
            Toggle Notification Banner
          </button>
          <div
            class="content-container-fill"
          >
            <div
              class="header"
            >
              <div
                class="alert-base error narrow"
                data-terra-application-banner="error"
                role="alert"
              >
                <div
                  class="body body-std"
                >
                  <span
                    class="icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="tui-Icon icon IconError"
                      focusable="false"
                      height="1em"
                      viewBox="0 0 48 48"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M24 0c13.3 0 24 10.7 24 24S37.3 48 24 48 0 37.3 0 24C0 10.8 10.7 0 23.9 0h.1z"
                        fill="#E50000"
                      />
                      <path
                        d="M7 21.5h34v5H7v-5z"
                        fill="#FFF"
                      />
                    </svg>
                  </span>
                  <div
                    class="section"
                  >
                    <strong
                      class="title"
                    >
                      <span>
                        Error.
                      </span>
                    </strong>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="main"
            >
              <div
                class="scroll normalizer"
              >
                <div>
                  <button
                    data-testid="show-2nd-banner"
                    type="button"
                  >
                    Toggle Notification Banner
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`NotificationBannerProvider Nested Notification Banner Provider unregisters banner with top-level Banner Provider 1`] = `
<div>
  <div
    class="content-container-fill"
  >
    <div
      class="header"
    >
      <div
        class="alert-base error narrow"
        data-terra-application-banner="error"
        role="alert"
      >
        <div
          class="body body-std"
        >
          <span
            class="icon"
          >
            <svg
              aria-hidden="true"
              class="tui-Icon icon IconError"
              focusable="false"
              height="1em"
              viewBox="0 0 48 48"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M24 0c13.3 0 24 10.7 24 24S37.3 48 24 48 0 37.3 0 24C0 10.8 10.7 0 23.9 0h.1z"
                fill="#E50000"
              />
              <path
                d="M7 21.5h34v5H7v-5z"
                fill="#FFF"
              />
            </svg>
          </span>
          <div
            class="section"
          >
            <strong
              class="title"
            >
              <span>
                Error.
              </span>
            </strong>
          </div>
        </div>
      </div>
    </div>
    <div
      class="main"
    >
      <div
        class="scroll normalizer"
      >
        <div>
          <button
            data-testid="show-banner"
            type="button"
          >
            Toggle Notification Banner
          </button>
          <div
            class="content-container-fill"
          >
            <div
              class="header"
            >
              <div
                class="alert-base error narrow"
                data-terra-application-banner="error"
                role="alert"
              >
                <div
                  class="body body-std"
                >
                  <span
                    class="icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="tui-Icon icon IconError"
                      focusable="false"
                      height="1em"
                      viewBox="0 0 48 48"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M24 0c13.3 0 24 10.7 24 24S37.3 48 24 48 0 37.3 0 24C0 10.8 10.7 0 23.9 0h.1z"
                        fill="#E50000"
                      />
                      <path
                        d="M7 21.5h34v5H7v-5z"
                        fill="#FFF"
                      />
                    </svg>
                  </span>
                  <div
                    class="section"
                  >
                    <strong
                      class="title"
                    >
                      <span>
                        Error.
                      </span>
                    </strong>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="main"
            >
              <div
                class="scroll normalizer"
              >
                <div>
                  <button
                    data-testid="show-2nd-banner"
                    type="button"
                  >
                    Toggle Notification Banner
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`NotificationBannerProvider Nested Notification Banner Provider unregisters banner with top-level Banner Provider 2`] = `
<div>
  <div
    class="content-container-fill"
  >
    <div
      class="header"
    />
    <div
      class="main"
    >
      <div
        class="scroll normalizer"
      >
        <div>
          <button
            data-testid="show-banner"
            type="button"
          >
            Toggle Notification Banner
          </button>
          <div
            class="content-container-fill"
          >
            <div
              class="header"
            >
              <div
                class="alert-base error narrow"
                data-terra-application-banner="error"
                role="alert"
              >
                <div
                  class="body body-std"
                >
                  <span
                    class="icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="tui-Icon icon IconError"
                      focusable="false"
                      height="1em"
                      viewBox="0 0 48 48"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M24 0c13.3 0 24 10.7 24 24S37.3 48 24 48 0 37.3 0 24C0 10.8 10.7 0 23.9 0h.1z"
                        fill="#E50000"
                      />
                      <path
                        d="M7 21.5h34v5H7v-5z"
                        fill="#FFF"
                      />
                    </svg>
                  </span>
                  <div
                    class="section"
                  >
                    <strong
                      class="title"
                    >
                      <span>
                        Error.
                      </span>
                    </strong>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="main"
            >
              <div
                class="scroll normalizer"
              >
                <div>
                  <button
                    data-testid="show-2nd-banner"
                    type="button"
                  >
                    Toggle Notification Banner
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`NotificationBannerProvider Nested Notification Banner Provider unregisters banner with top-level Banner Provider 3`] = `
<div>
  <div
    class="content-container-fill"
  >
    <div
      class="header"
    />
    <div
      class="main"
    >
      <div
        class="scroll normalizer"
      >
        <div>
          <button
            data-testid="show-banner"
            type="button"
          >
            Toggle Notification Banner
          </button>
          <div
            class="content-container-fill"
          >
            <div
              class="header"
            />
            <div
              class="main"
            >
              <div
                class="scroll normalizer"
              >
                <div>
                  <button
                    data-testid="show-2nd-banner"
                    type="button"
                  >
                    Toggle Notification Banner
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`NotificationBannerProvider registerNotificationBanner registers banner when rendered after mount 1`] = `
<div>
  <div
    class="content-container-fill"
  >
    <div
      class="header"
    />
    <div
      class="main"
    >
      <div
        class="scroll normalizer"
      >
        <div>
          <button
            data-testid="show-banner"
            type="button"
          >
            Toggle Notification Banner
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`NotificationBannerProvider registerNotificationBanner registers banner when rendered after mount 2`] = `
<div>
  <div
    class="content-container-fill"
  >
    <div
      class="header"
    >
      <div
        class="alert-base error narrow"
        data-terra-application-banner="error"
        role="alert"
      >
        <div
          class="body body-std"
        >
          <span
            class="icon"
          >
            <svg
              aria-hidden="true"
              class="tui-Icon icon IconError"
              focusable="false"
              height="1em"
              viewBox="0 0 48 48"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M24 0c13.3 0 24 10.7 24 24S37.3 48 24 48 0 37.3 0 24C0 10.8 10.7 0 23.9 0h.1z"
                fill="#E50000"
              />
              <path
                d="M7 21.5h34v5H7v-5z"
                fill="#FFF"
              />
            </svg>
          </span>
          <div
            class="section"
          >
            <strong
              class="title"
            >
              <span>
                Error.
              </span>
            </strong>
          </div>
        </div>
      </div>
    </div>
    <div
      class="main"
    >
      <div
        class="scroll normalizer"
      >
        <div>
          <button
            data-testid="show-banner"
            type="button"
          >
            Toggle Notification Banner
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`NotificationBannerProvider renders with children 1`] = `
<div>
  <div
    class="content-container-fill"
  >
    <div
      class="header"
    />
    <div
      class="main"
    >
      <div
        class="scroll normalizer"
      />
    </div>
  </div>
</div>
`;

exports[`NotificationBannerProvider renders with children 2`] = `
<div>
  <div
    class="content-container-fill"
  >
    <div
      class="header"
    />
    <div
      class="main"
    >
      <div
        class="scroll normalizer"
      >
        <div>
          child
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`NotificationBannerProvider renders with children that has a an banner 1`] = `
<div>
  <div
    class="content-container-fill"
  >
    <div
      class="header"
    >
      <div
        class="alert-base error narrow"
        data-terra-application-banner="error"
        role="alert"
      >
        <div
          class="body body-std"
        >
          <span
            class="icon"
          >
            <svg
              aria-hidden="true"
              class="tui-Icon icon IconError"
              focusable="false"
              height="1em"
              viewBox="0 0 48 48"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M24 0c13.3 0 24 10.7 24 24S37.3 48 24 48 0 37.3 0 24C0 10.8 10.7 0 23.9 0h.1z"
                fill="#E50000"
              />
              <path
                d="M7 21.5h34v5H7v-5z"
                fill="#FFF"
              />
            </svg>
          </span>
          <div
            class="section"
          >
            <strong
              class="title"
            >
              <span>
                Error.
              </span>
            </strong>
          </div>
        </div>
      </div>
    </div>
    <div
      class="main"
    >
      <div
        class="scroll normalizer"
      >
        <div>
          <button
            data-testid="show-banner"
            type="button"
          >
            Toggle Notification Banner
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`NotificationBannerProvider unregisterNotificationBanner unregisters notification banner when id is provided 1`] = `
<div>
  <div
    class="content-container-fill"
  >
    <div
      class="header"
    >
      <div
        class="alert-base error narrow"
        data-terra-application-banner="error"
        role="alert"
      >
        <div
          class="body body-std"
        >
          <span
            class="icon"
          >
            <svg
              aria-hidden="true"
              class="tui-Icon icon IconError"
              focusable="false"
              height="1em"
              viewBox="0 0 48 48"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M24 0c13.3 0 24 10.7 24 24S37.3 48 24 48 0 37.3 0 24C0 10.8 10.7 0 23.9 0h.1z"
                fill="#E50000"
              />
              <path
                d="M7 21.5h34v5H7v-5z"
                fill="#FFF"
              />
            </svg>
          </span>
          <div
            class="section"
          >
            <strong
              class="title"
            >
              <span>
                Error.
              </span>
            </strong>
          </div>
        </div>
      </div>
    </div>
    <div
      class="main"
    >
      <div
        class="scroll normalizer"
      >
        <div>
          <button
            data-testid="show-banner"
            type="button"
          >
            Toggle Notification Banner
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`NotificationBannerProvider unregisterNotificationBanner unregisters notification banner when id is provided 2`] = `
<div>
  <div
    class="content-container-fill"
  >
    <div
      class="header"
    />
    <div
      class="main"
    >
      <div
        class="scroll normalizer"
      >
        <div>
          <button
            data-testid="show-banner"
            type="button"
          >
            Toggle Notification Banner
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
